src/main/java/org/cloudfoundry/identity/uaa/api/group/UaaGroupOperations.java

```diff
@@ -81,7 +85,7 @@
 	 * @return The page of groups.
 	 * @see org.cloudfoundry.identity.uaa.api.common.model.expr.FilterRequestBuilder
 	 */
-	public PagedResult<UaaGroup> getGroups(FilterRequest request);
+	public ScimGroups getGroups(FilterRequest request);
```

Comment:
Is this [example](https://github.com/cloudfoundry/uaa/blob/develop/uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/util/MockMvcUtils.java#L218-224) what you are looking for? Here I am able to type reference the search results that return groups, and deserialize it using JSON mapper